incubator/binding-openapi/src/main/java/io/aklivity/zilla/runtime/binding/openapi/internal/streams/OpenapiClientFactory.java

```diff
@@ -0,0 +1,1065 @@
+    }
+
+    @Override
+    public void attach(
+        BindingConfig binding)
+    {
+        OpenapiBindingConfig openapiBinding = new OpenapiBindingConfig(binding);
+        bindings.put(binding.id, openapiBinding);
+    }
+
+    @Override
+    public void detach(
+        long bindingId)
+    {
+        bindings.remove(bindingId);
+    }
+
+    @Override
+    public MessageConsumer newStream(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length,
+        MessageConsumer receiver)
+    {
+        final BeginFW begin = beginRO.wrap(buffer, index, index + length);
+        final long originId = begin.originId();
+        final long routedId = begin.routedId();
+        final long initialId = begin.streamId();
+        final long affinity = begin.affinity();
+        final long authorization = begin.authorization();
+        final OctetsFW extension = begin.extension();
+        final OpenapiBeginExFW openapiBeginEx = extension.get(openapiBeginExRO::tryWrap);
+
+        final OpenapiBindingConfig binding = bindings.get(routedId);
+
+        MessageConsumer newStream = null;
+
+        if (binding != null && binding.isCompositeBinding(originId))
+        {
+            final String operationId = openapiBeginEx.operationId().asString();
+
+            OpenapiRouteConfig route = binding.resolve(authorization);
+
+            if (route != null)
+            {
+                newStream = new OpenapiStream(
+                    receiver,
+                    originId,
+                    routedId,
+                    initialId,
+                    affinity,
+                    authorization,
+                    route.id,
+                    operationId)::onOpenapiMessage;
+            }
+
+        }
+
+        return newStream;
+    }
+
+    private MessageConsumer newStream(
+        MessageConsumer sender,
+        long originId,
+        long routedId,
+        long streamId,
+        long sequence,
+        long acknowledge,
+        int maximum,
+        long traceId,
+        long authorization,
+        long affinity,
+        Flyweight extension)
+    {
+        final BeginFW begin = beginRW.wrap(writeBuffer, 0, writeBuffer.capacity())
+                .originId(originId)
+                .routedId(routedId)
+                .streamId(streamId)
+                .sequence(sequence)
+                .acknowledge(acknowledge)
+                .maximum(maximum)
+                .traceId(traceId)
+                .authorization(authorization)
+                .affinity(affinity)
+                .extension(extension.buffer(), extension.offset(), extension.sizeof())
+                .build();
+
+        final MessageConsumer receiver =
+                streamFactory.newStream(begin.typeId(), begin.buffer(), begin.offset(), begin.sizeof(), sender);
+
+        receiver.accept(begin.typeId(), begin.buffer(), begin.offset(), begin.sizeof());
+
+        return receiver;
+    }
+
+    private void doBegin(
```

Comment:
Let's move these pseudo-static private `newStream` and `doXXX` methods to the end of the factory for improved readability so that public interface method override `newStream` is next to the constructor of the `HttpStream` inner class.